### PR TITLE
Build definition renames for FreeBSD

### PR DESCRIPTION
**Pull request by dagood**
`Build_Linux_x64_FreeBSD` => `Build_FreeBSD_x64`

`linux-build.yml` => `bash-build.yml` (Definitely open to suggestions on this one....)

Followup from https://github.com/dotnet/core-setup/pull/4936#discussion_r246409102.

/cc @wfurt @bartonjs

### CODE REVIEW

File: eng/jobs/finalize-publish.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
                   succeeded('Build_Linux_Arm64'),
                   succeeded('Build_Linux_Arm'),
                   succeeded('Build_Linux_x64_Alpine36'),
-                  succeeded('Build_Linux_x64_FreeBSD'),
+                  succeeded('Build_FreeBSD_x64'),
```
Comment (bartonjs) [inline eng/jobs/finalize-publish.yml]: I'd fix the sorting on the two changes in this file.

File: eng/jobs/bash-build.yml [hunk 2]
```diff
@@ -22,24 +22,24 @@ jobs:
   - job: ${{ parameters.displayName }}
     displayName: ${{ parameters.displayName }}
     pool:
-      ${{ if and(eq(parameters.displayName, 'Build_Linux_x64_FreeBSD'), ne(variables['System.TeamProject'], 'public')) }}:
+      ${{ if and(eq(parameters.displayName, 'Build_FreeBSD_x64'), ne(variables['System.TeamProject'], 'public')) }}:
```
Comment (bartonjs) [inline eng/jobs/bash-build.yml]: (Up on line 8)  does `osGroup` need to be changed to FreeBSD for this group?

Comment (dagood) [reply eng/jobs/bash-build.yml]: OSGroup does need to be FreeBSD, but the value in this file is only the default--it's already overridden here:

https://github.com/dotnet/core-setup/blob/c5f6ce6670ba67cf1e5aaec87c1053aa4fc62bf9/azure-pipelines.yml#L134-L138

File: eng/jobs/finalize-publish.yml (hunk 1 above)
Comment (dagood) [reply eng/jobs/finalize-publish.yml]: Hmm, noticed the non-alpha order was in sync with `azure-pipelines.yml`. I'm going to sort that too.

**Pull request merged**